### ISSUE TITLE
Inconvenient `NonEmptyString`

### ISSUE THREAD

**Issue by pawelprazak** [VirtusLab/besom#138]
Currently, the API forces to explicitly type string `val`'s, e.g.:
```scala
val bucketName: NonEmptyString = "myResource"
```

It would be nice to find a way to solve this without sacrificing the type-safety.

**Comment by lbialy** [comment]
Solved by #137

**Comment by pawelprazak** [comment]
Using `"enable-" + api.replace(".", "-"),` should behave like `s"enable-${ api.replace(".", "-") }"`, got:

```
Only constant strings or string interpolations are allowed here, use NonEmptyString.apply instead!
```

**Comment by lbialy** [comment]
Hopefully closed by #375!

**Comment by lbialy** [comment]
Reopen if any other problems are found.
